src/main/java/org/springframework/integration/dsl/IntegrationFlowDefinition.java

```diff
@@ -2346,7 +2347,7 @@ public void accept(GenericEndpointSpec<BridgeHandler> bridge) {
 			}
 		}
 		if (hasSubFlows) {
-			channel(afterRouterChannel);
+			handle(bridgeHandler);
```

Comment:
From one side I protected us from the undesired tickets: https://github.com/spring-projects/spring-integration-java-dsl/wiki/Spring-Integration-Java-DSL-Reference#subflows.
From another: I see only two ways to push messages from sub-flows to the top - channel and bridge.

And I'm not sure in that "lazily created bridge" because it's enough difficult here to be consistent, but in the end that would be more complicated...